File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [];
    
    for (let x = 1; x <= Math.sqrt(yellow); x++) {
        y = yellow / x;
        
        if (((x+2)*2 + y*2) === brown) {
            answer.push(y+2);
            answer.push(x+2);
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for yh in range(1, int(yellow**0.5)+1):
        if yellow % yh == 0:
            yw = yellow // yh

            if (brown == (yw+2)*2 + yh*2):
                answer = [yw+2, yh+2] 
                break

    return answer